proof/refine/RISCV64/CSpace1_R.thy

```diff
@@ -427,17 +411,17 @@ proof -
   } note x = this
   from assms
   show ?thesis
-  apply (cases c)
-            apply (simp_all add: simps)[5]
-       defer
-       apply (simp_all add: simps)[4]
+  apply (cases c; simp add: simps)
+   defer
    apply (clarsimp simp: simps the_arch_cap_def)
    apply (rename_tac arch_cap)
-   apply (case_tac arch_cap; simp add: simps arch_update_cap_data_def
-                             RISCV64_H.updateCapData_def)
+   apply (case_tac arch_cap)
+        apply (simp_all add: arch_update_cap_data_def
+                             RISCV64_H.updateCapData_def)[5]
```

Comment:
`simp_all` is a bit old-style, these days I would prefer something like
```isabelle
   apply (case_tac arch_cap; simp add: arch_update_cap_data_def RISCV64_H.updateCapData_def)
```
This doesn't need the explicit goal limiter and won't leak if the number of goals change.